kudo-operator/templates/spark-history-server-deployment.yaml

```diff
@@ -0,0 +1,64 @@
+{{- if .Params.enableHistoryServer }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: history-server
+  namespace: {{ .Namespace }}
+  labels:
+    app.kubernetes.io/name: history-server
+    app.kubernetes.io/instance: {{ .Name }}
+    app.kubernetes.io/version: {{ .Params.operatorVersion }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: history-server
+      app.kubernetes.io/instance: {{ .Name }}
+      app.kubernetes.io/version: {{ .Params.operatorVersion }}
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: history-server
+        app.kubernetes.io/instance: {{ .Name }}
+        app.kubernetes.io/version: {{ .Params.operatorVersion }}
+    spec:
+      {{- if .Params.createOperatorServiceAccount }}
+      serviceAccountName: {{ .Name }}-{{ .Params.operatorServiceAccountName }}
+      {{- else }}
+      serviceAccountName: {{ .Params.operatorServiceAccountName }}
+      {{- end }}
+      containers:
+      - name: history-server
+        image: {{ .Params.operatorImageName }}:{{ .Params.operatorVersion }}
+        imagePullPolicy: {{ .Params.imagePullPolicy }}
+        command: ["/opt/spark/sbin/start-history-server.sh"]
+        env:
+        - name: SPARK_NO_DAEMONIZE
+          value: "true"
+        - name: AWS_ACCESS_KEY_ID
+          valueFrom:
+            secretKeyRef:
+              name: s3-secret
+              key: accessKeyID
```

Comment:
I'm afraid, this is not going to work in case the secret is not present and Spark History is enabled